src/nbl/ext/ImGui/shaders/fragment.hlsl

```diff
@@ -1,9 +1,41 @@
+#ifndef NBL_TEXTURES_BINDING_IX
+#error "NBL_TEXTURES_BINDING_IX must be defined!"
+#endif
+
+#ifndef NBL_SAMPLER_STATES_BINDING_IX
+#error "NBL_SAMPLER_STATES_BINDING must be defined!"
+#endif
+
+#ifndef NBL_TEXTURES_SET_IX
+#error "NBL_TEXTURES_SET_IX must be defined!"
+#endif
+
+#ifndef NBL_SAMPLER_STATES_SET_IX
+#error "NBL_SAMPLER_STATES_SET_IX must be defined!"
+#endif
+
+#ifndef NBL_RESOURCES_COUNT
+#error "NBL_RESOURCES_COUNT must be defined!"
+#endif
+
 #include "common.hlsl"
 
-[[vk::combinedImageSampler]][[vk::binding(0, 0)]] Texture2D sampleTexture : register(t0);
-[[vk::combinedImageSampler]][[vk::binding(0, 0)]] SamplerState linearSampler : register(s0);
+[[vk::push_constant]] struct PushConstants pc;
+
+// separable image samplers to handle textures we do descriptor-index
+[[vk::binding(NBL_TEXTURES_BINDING_IX, NBL_TEXTURES_SET_IX)]] Texture2D textures[NBL_RESOURCES_COUNT];
+[[vk::binding(NBL_SAMPLER_STATES_BINDING_IX, NBL_SAMPLER_STATES_SET_IX)]] SamplerState samplerStates[NBL_RESOURCES_COUNT];
```

Comment:
there should be a sampler array size thats separate from the texture array size

Reply:
so we want accessors? Texture ID which is basically ID of those arrays makes texture map to a sampler now

Reply:
no just indep sizes